verifier/plonk/gates/random_access.go

```diff
@@ -0,0 +1,156 @@
+package gates
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/Electron-Labs/plonky2-groth16-verifier/goldilocks"
+	"github.com/consensys/gnark/frontend"
+)
+
+type RandomAccessGate struct {
+	Bits              int `json:"bits"`
+	NumCopies         int `json:"num_copies"`
+	NumExtraConstants int `json:"num_extra_constants"`
+}
+
+func NewRandomAccessGate(id string) *RandomAccessGate {
+	id = strings.Split(id, ", _phantom")[0]
```

Comment:
I think we should have an equality check here that the split string is actually ": PhantomData<plonky2_field::goldilocks_field::GoldilocksField> }<D=2>"